src/crescendo.ts

```diff
@@ -54,59 +78,55 @@ export class Crescendo extends Note {
 
     // The height at the open end of the cresc/decresc
     this.height = 15;
-
-    Vex.Merge(this.render_options, {
-      // Extensions to the length of the crescendo on either side
-      extend_left: 0,
-      extend_right: 0,
-      // Vertical shift
-      y_shift: 0,
-    });
   }
 
   // Set the line to center the element on
-  setLine(line) {
+  setLine(line: number): this {
     this.line = line;
     return this;
   }
 
   // Set the full height at the open end
-  setHeight(height) {
+  setHeight(height: number): this {
     this.height = height;
     return this;
   }
 
   // Set whether the sign should be a descresendo by passing a bool
   // to `decresc`
-  setDecrescendo(decresc) {
+  setDecrescendo(decresc: boolean): this {
     this.decrescendo = decresc;
     return this;
   }
 
   // Preformat the note
-  preFormat() {
+  preFormat(): this {
     this.preFormatted = true;
     return this;
   }
 
   // Render the Crescendo object onto the canvas
-  draw() {
-    this.checkContext();
+  draw(): void {
+    const ctx = this.checkContext();
+    if (!this.stave) throw new Vex.RERR('NoStave', "Can't draw without a stave.");
     this.setRendered();
 
     const tick_context = this.getTickContext();
     const next_context = TickContext.getNextContext(tick_context);
 
     const begin_x = this.getAbsoluteX();
-    const end_x = next_context ? next_context.getX() : this.stave.x + this.stave.width;
+    const end_x = next_context ? next_context.getX() : this.stave.getX() + this.stave.getWidth();
     const y = this.stave.getYForLine(this.line + -3) + 1;
 
     L('Drawing ', this.decrescendo ? 'decrescendo ' : 'crescendo ', this.height, 'x', begin_x - end_x);
 
-    renderHairpin(this.context, {
-      begin_x: begin_x - this.render_options.extend_left,
-      end_x: end_x + this.render_options.extend_right,
-      y: y + this.render_options.y_shift,
+    renderHairpin(ctx, {
+      // eslint-disable-next-line
```

Comment:
Unclear why you have these eslint comments?

Reply:
I forgot to remve them when I removed the `!` operator.

Done!